arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Raise.kt

```diff
@@ -238,6 +239,14 @@ public interface Raise<in Error> {
     is Either.Right -> value
   }
 
+  @RaiseDSL
+  public fun <A> Ior<Error, A>.bind(): A =
```

Comment:
I came to the idea of doing this from studying why `either { recover({ ... }) { ... } }` works correctly. This ensures that in the recover block we hit this `bind()` if we reference `bind()` vs the one on `IorScope`. This allows `recover` to work in the way it does for either. However, the downside of this is, we do nothing with `Ior.Both`'s left value which may be undesirable.

Originally I tried adding a new `recover(...)` method to `IorRaise` but couldn't quite get that to work right without more major changes. However, I might have missed something there.

Furthermore, the issue identified with `Ior` will happen with any other data type that has its own Raise variant. So `NullableRaise` `OptionRaise` etc. I didn't want to start making changes for that as I wanted to keep the discussion focused on the core problem. However, it's worth considering on this PR for sure.